x/feegrant/keeper/grpc_query.go

```diff
@@ -110,7 +110,8 @@ func (q Keeper) AllowancesByGranter(c context.Context, req *feegrant.QueryAllowa
 	var grants []*feegrant.Grant
 
 	store := ctx.KVStore(q.storeKey)
-	pageRes, err := query.Paginate(store, req.Pagination, func(key []byte, value []byte) error {
+	prefixStore := prefix.NewStore(store, feegrant.FeeAllowanceKeyPrefix)
+	pageRes, err := query.Paginate(prefixStore, req.Pagination, func(key []byte, value []byte) error {
```

Comment:
This is the reasons you are mentioning in https://github.com/cosmos/cosmos-sdk/pull/10830/files#r796502828, right?

I think this PR generally looks good, but IMO it's very hard to read because of the `[1:]` it introduces.

Can we revert `ParseAddressesFromFeeAllowanceKey`, and here on L166 do:

```go
// ParseAddressesFromFeeAllowanceKey expects the full key including the prefix.
granter, _ := feegrant.ParseAddressesFromFeeAllowanceKey(append([]byte{FeeAllowanceKeyPrefix}, key))
```